Added antialias setting to DynamicFont

ivodopiviz opened the pull request:
And added editor settings for the code editor while at it. Maybe GIF wasn't the best idea to showcase it.

![capture](https://user-images.githubusercontent.com/250464/37002535-53cf0846-20ca-11e8-81b5-473bd54bfd1b.gif)

To be honest this is more of a request for comments than a PR: 

* I'm not happy with the category of the setting. It makes sense in the gui, but it's not really a theme setting. I'm also unsure how the Font setting on that same section works, it looks nothing like the editor fonts part. The lack of size setting also tells me this isn't the right place.

* I'm not sure if the antialias setting should be bound. #17094 seems to do it for hinting so I guess the answer is yes?

Other than that, it does what the original issue reporter wanted, so with the necessary adjustments it's pretty much done.

Calinou commented:
> I'm also unsure how the Font setting on that same section works, it looks nothing like the editor fonts part. The lack of size setting also tells me this isn't the right place.

The font size setting is in **Editor** → **Interface**. The font anti-aliasing settings should probably be moved there.

ivodopiviz commented:
Ok, moved setting around and squashed commit to see if that fixed the previously broken travis. Alas, it does not.

nooperation commented on scene/resources/dynamic_font.h:
```diff
@@ -86,6 +87,8 @@ class DynamicFontData : public Resource {
 	String get_font_path() const;
 	void set_force_autohinter(bool p_force);
+	void set_use_antialias(bool p_use);
+	bool get_use_antialias();
```
can this be const like get_font_path?

ivodopiviz commented:
@nooperation Yes we can, good catch :)

Updated and rebased.

akien-mga commented on scene/resources/dynamic_font.cpp:
```diff
@@ -79,13 +79,20 @@ void DynamicFontData::set_force_autohinter(bool p_force) {
 void DynamicFontData::_bind_methods() {
 	ClassDB::bind_method(D_METHOD("set_font_path", "path"), &DynamicFontData::set_font_path);
 	ClassDB::bind_method(D_METHOD("get_font_path"), &DynamicFontData::get_font_path);
+	ClassDB::bind_method(D_METHOD("set_use_antialias", "antialias"), &DynamicFontData::set_use_antialias);
```
The more idiomatic bindings would be: `"set_use_antialias", "enabled"` and `"is_using_antialias"`.

ivodopiviz commented:
Boy, did I screw up that merge. Anyway, I re-merged with master and condensed everything in one commit. Only changed the bindings however, I wanted to keep the setter and getter with set and get in their names.

reduz commented:
antialias setting should most likely go in DynamicFontData::CacheID since, as far as I know, its a character generation option, hence another texture may be needed.

akien-mga commented:
> antialias setting should most likely go in DynamicFontData::CacheID since, as far as I know, its a character generation option, hence another texture may be needed.

Bump.

ivodopiviz commented:
I've been a little busy and couldn't get to this before. I'll take a look.

ivodopiviz committed:
Added antialias toggling to DynamicFont

Made it possible to deactivate antialias in DynamicFonts. Added editor setting for it.
Added bindings and doc. Fixes #883

ivodopiviz committed:
Moved antialias to DynamicFont

Also made it use the cache system.

ivodopiviz commented:
@reduz Ok, I'm kinda stuck. I moved the code to DynamicFont itself and used the cacheid in a similar way as filter or mipmap work, is this what you meant?

It's currently not working properly, behavior is erratic but I'm most certainly misunderstanding how the whole thing works, but I'd like to know if I'm at least going in the right direction 😄

aaronfranke commented:
Would fix https://github.com/godotengine/godot/pull/17292

akien-mga commented:
Superseded by #23645.
